src/lib/hooks/useNotificationHandler.ts

```diff
@@ -0,0 +1,207 @@
+                screen: 'Profile',
+                params: {
+                  name: uri.host,
+                },
+              })
+            }, 500)
+            break
+          case 'mention':
+          case 'reply':
+            const urip = new AtUri(payload.uri)
+            setTimeout(() => {
+              // @ts-expect-error types are weird here
+              navigation.navigate('HomeTab', {
+                screen: 'PostThread',
+                params: {
+                  name: urip.host,
+                  rkey: urip.rkey,
+                },
+              })
+            }, 500)
+            break
+        }
+      }
+    }
+
+    Notifications.setNotificationHandler({
+      handleNotification: async e => {
+        if (e.request.trigger.type !== 'push') return DEFAULT_HANDLER_OPTIONS
+
+        logger.debug(
+          'Notifications: received',
+          {e},
+          logger.DebugContext.notifications,
+        )
+
+        const payload = e.request.trigger.payload as NotificationRecord
+        if (
+          payload.reason === 'chat-message' &&
+          payload.recipientDid === currentAccount?.did &&
+          currentConvoId !== payload.convoId
+        ) {
+          return {
+            shouldShowAlert: true,
+            shouldPlaySound: false,
+            shouldSetBadge: false,
+          }
+        }
+
+        // Any notification other than a chat message should invalidate the unread page
+        invalidateCachedUnreadPage()
+        return DEFAULT_HANDLER_OPTIONS
+      },
+    })
+
+    const responseReceivedListener =
+      Notifications.addNotificationResponseReceivedListener(e => {
+        logger.debug(
+          'Notifications: response received',
+          {
+            actionIdentifier: e.actionIdentifier,
+          },
+          logger.DebugContext.notifications,
+        )
+
+        if (
+          e.actionIdentifier === Notifications.DEFAULT_ACTION_IDENTIFIER &&
+          e.notification.request.trigger.type === 'push'
+        ) {
+          logger.debug(
+            'User pressed a notification, opening notifications tab',
+            {},
+            logger.DebugContext.notifications,
+          )
+          track('Notificatons:OpenApp')
+          logEvent('notifications:openApp', {})
+          invalidateCachedUnreadPage()
+          truncateAndInvalidate(queryClient, RQKEY_NOTIFS())
+          logger.debug('Notifications: handleNotification', {
+            content: e.notification.request.content,
+            payload: e.notification.request.trigger.payload,
+          })
+          handleNotification(
+            e.notification.request.trigger.payload as NotificationRecord,
+          )
+          Notifications.dismissAllNotificationsAsync()
+        }
+      })
+
+    if (
+      storedPayload?.reason === 'chat-message' &&
+      currentAccount?.did === storedPayload.recipientDid
+    ) {
+      handleNotification(storedPayload)
+      storedPayload = undefined
+    }
```

Comment:
When we encounter a case where you open a notification that doesn't belong to the currently signed in account, we switch the account above and store the payload. Once `currentAccount` changes, we will re-handle the notification.

This only applies to `chat-message` right now because we don't have a `recipientDid` yet on other notification payloads.

Reply:
Nice one